app/api/v1/line.py

```
@@ -116,24 +110,65 @@ async def handle_line_callback(
+    try:
+        signature = request.headers.get("X-Line-Signature")
+        if not signature:
+            raise HTTPException(
+                status_code=400, detail="X-Line-Signature header is missing"
+            )
+
+        body = await request.body()
+        return signature, body.decode()
+    except ClientDisconnect:
+        logger.warning("Client disconnected while reading request body")
+        raise HTTPException(status_code=400, detail="Client disconnected")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)